examples/upgrade_forseti_with_v5.0/tutorial.md

```diff
@@ -0,0 +1,245 @@
+  [Terraform command-line interface](https://www.terraform.io/downloads.html)
+  in the 0.12 series.
+- The ID of the GCP project in which Forseti is deployed.
+- A service account in the organization with the
+  [roles required by the Terraform module](https://registry.terraform.io/modules/terraform-google-modules/forseti/google/4.3.0#iam-roles).
+- A
+  [JSON key file](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys)
+  for the service account.
+- If you are an Org Admin in the organization in which you deploying Forseti, a separate Service Account and Key are recommended,
+  but not required.
+- **Strongly recommended out of an overabundance of caution:** A backup of your current state.
+  - In the Forseti Server's GCS Bucket
+    - Scanner rules
+    - Server config file
+    - Scanner Violations
+    - Inventory summary
+  - [CloudSQL database](https://cloud.google.com/sql/docs/mysql/backup-recovery/backups)
+
+## Configuring Terraform
+
+Terraform can assume the identity of a service account through a
+strategy called
+[Application Default Credentials](https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application)
+when provisioning resources. To enable this approach, set the
+appropriate environment variable to the path of the service account JSON
+key file:
+
+```sh
+export GOOGLE_APPLICATION_CREDENTIALS="PATH_TO_JSON_KEY_FILE"
+```
+As stated in the pre-requisites, if you have Org Admin privilges, you do not need to complete this step.
+
+## Backup the Terraform State
+In a shell, navigate to the folder containing your user-defined Terraform module, most likely in a **main.tf**.
+
+The following command will simultaneously backup your existing Terraform state and remove resource from the current state file.  This will not affect your existing Forseti deployment.
+```sh
+terraform state rm $(terraform state list)
+```
+## Update main.tf
+In order to support this upgrade, we'll need to update a few input variables.
+
+### Source
+The **source** will need to point to the root terraform-google-forseti module.
+```
+source = "terraform-google-modules/forseti/google"
+```
+If you have cloned the module to your local file system, you may set the **source** path to
+the directory containing the module.
+
+### Version
+The **version** will need to be 5.0.0.
+```
+version = "5.0.0"
+```
+If you have set the **source** path to the directory containing the module, omit the **version** variable.
+
+### Region
+If you have a **region**, it will need to be split into the cloudsql_region, server_region, and client_region
+variables.
+
+**NOTE:** In order to prevent data loss to your CloudSQL database, please double check the region where
+your CloudSQL instance currently exists and update the **cloudsql_region** variable, accourdingly.
+
+Before (example):
+```
+region = "us-central1"
+```
+After (example):
+```
+cloudsql_region = "us-central1"
```

Comment:
Should be mention that these need to be added to the `terraform.tfvars` and the variables.tf` files?

Reply:
It depends on how people have Terraform setup. You can either put all of this in the main.tf or use variables.tf and the terraform.tfvars. I think this is a user preference for how to setup.